JSIL/ILBlockTranslator.cs

```diff
@@ -1130,11 +1138,28 @@ bool forDynamic
             var translated = TranslateNode(node as dynamic);
 
             var statement = translated as JSStatement;
-            if (statement == null) {
-                var expression = (JSExpression)translated;
+            if (statement == null)
+            {
+                var expression = (JSExpression) translated;
 
                 if (expression != null)
+                {
+                    var ranges =
+                        node.GetSelfAndChildrenRecursive<ILExpression>(item => true).SelectMany(item => item.ILRanges);
```

Comment:
Will look on it later - really this point will change a lot while Source map support will be more mature.